### PR TITLE
WIP: array notation for forms

### PR DESCRIPTION
**Pull request by ryanbethel**
The goal is to normalize form data into JavaScript on the server or on the client consistently.
- Adds support for Floats in addition to Integers.
- Adds support for nested arrays in form notation. `'user.addr[1].firstname'` and `'foo[]'`
- Adds support for FormData objects for normalizing form data on the client.

### CODE REVIEW

File: src/convert-to-nested-object.js [hunk 1]
```diff
@@ -1,35 +1,160 @@
...
+    return { path: entry.path.replace(/\[]$/, ''), value: entry.value }
   }
+  return entry
+}
 
-  let str = arr[0],
-    last = obj,
-    props,
-    valProp
+// for iterable objects like FormData this adds the [] inticator at the end of duplicate key
```
Comment (macdonst) [inline src/convert-to-nested-object.js:32]: inticator -> indicator

**Review by macdonst — APPROVED** [review]
LGTM! I made two minor comments that you may want to clean up as you edit the README file but go ahead and merge when you are ready.

File: src/convert-to-nested-object.js (hunk 1 above)
Comment (ryanbethel) [reply src/convert-to-nested-object.js]: done

File: src/convert-to-nested-object.js [hunk 2]
```diff
@@ -1,35 +1,160 @@
...
+  return [ currentArray[order], val => (currentArray[order] = val) ]
 }
 
-export default function convertToNestedObject (data) {
-  let obj = {}
-  Object.entries(data).forEach(entry => convertExtend(entry, obj))
-  return obj
+export default function parseFormData ( formData, { removeEmptyString = false, transformEntry = false, duplicateKeys = [] } = {}){
+  let result = {}
+  // let flatten = [ 'zoo' ]
```
[the remark this answers is not in the document]
Comment (ryanbethel) [reply src/convert-to-nested-object.js]: done

**Pull request merged**